sql/mysqld.cc

```diff
@@ -6481,10 +6454,10 @@ void handle_connections_sockets()
   while (!abort_loop)
   {
 #ifdef HAVE_POLL
-    retval= poll(fds, socket_count, -1);
+    retval= poll(fds.data(), fds.size(), -1);
 #else
     readFDs=clientFDs;
-    retval= select((int) 0,&readFDs,0,0,0);
+    retval= select(FD_SETSIZE, &readFDs, NULL, NULL, NULL);
```

Comment:
hmm. I don't even see that nfds==0 is documented to work.
okay

Reply:
> I don't even see that nfds==0 is documented to work.

Some older code use `select(0, NULL, NULL, NULL, &timeout);` as a sleep routine. And [documentation for select() on Windows](https://docs.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-select), mentions that they ignore `nfds`.

> is this branch even compiled on any platform?

I don't think it is compiled. When I tried to hide `poll()` support, some other code failed to compile. I guess, `poll()` support is effectively mandatory to compile MariaDB. But since I touch that code, I'd like to not break code hidden under ifdef's if possible.